thermal_base/thermal_base.py

```diff
@@ -126,14 +132,36 @@ def extract_temperatures_dji(self):
 
         ## calculating the raw sensor values
 
-        im = Image.open(self.image_path)
-        # concatenate APP3 chunks of data
-        a = im.applist[3][1]
-        for i in range(4, 14):
-            a += im.applist[i][1]
-        # create image from bytes
-        img = Image.frombytes("I;16L", (640, 512), a)
-
+        # OS and architecture checks
+        os_name = "windows" if "win" in sys.platform else "linux"
+        architecture_name = "release_x64" if "64bit" in platform.architecture()[0] else "release_x86"
+        dji_binary = "dji_irp.exe" if "win" in sys.platform else "dji_irp"
+        dji_executables_url = "https://dtpl-ai-public.s3.ap-south-1.amazonaws.com/Thermal_Image_Analysis/DJI_SDK/dji_executables.zip"
+
+        # If DJI executable files aren't present, download them and extract the folder contents.
+        if not Path('dji_executables').exists():
+            r = requests.get(dji_executables_url, stream=True)
+            z = zipfile.ZipFile(io.BytesIO(r.content))
+            z.extractall()
+
+        if "linux" in os_name:
+            # Linux needs path of libdirp.so file added to Environment variable and Execute permission to executable file.
+            path_executable = str(Path("./dji_executables", os_name, architecture_name))
+            os.environ['LD_LIBRARY_PATH'] = path_executable
+            sp.run(["chmod", "u+x", str(Path(path_executable, dji_binary))])
+        else:
```

Comment:
Can you check for windows explicitly? That way, someone running it in mac os won't enter this block